tests/snuba/api/endpoints/test_organization_events_span_indexed.py

```diff
@@ -2038,6 +2040,56 @@ def test_extrapolation(self) -> None:
         assert data[1]["count()"] == 1
         assert confidence[1]["count()"] in ("high", "low")
 
+    @mock.patch(
+        "sentry.utils.snuba_rpc._make_rpc_requests",
+        wraps=_make_rpc_requests,
+    )
+    def test_extrapolation_mode_server_only(self, mock_rpc_request: mock.MagicMock) -> None:
+        spans = []
+        spans.append(
+            self.create_span(
+                {
+                    "description": "foo",
+                    "sentry_tags": {"status": "success"},
+                    "measurements": {"server_sample_rate": {"value": 0.1}},
+                },
+                start_ts=self.ten_mins_ago,
+            )
+        )
+        spans.append(
+            self.create_span(
+                {
+                    "description": "bar",
+                    "sentry_tags": {"status": "success"},
+                },
+                start_ts=self.ten_mins_ago,
+            )
+        )
+        self.store_spans(spans, is_eap=True)
+        response = self.do_request(
+            {
+                "field": ["description", "count()"],
+                "orderby": "-count()",
+                "query": "",
+                "project": self.project.id,
+                "dataset": "spans",
+                "extrapolationMode": "serverOnly",
+            }
+        )
+
+        assert response.status_code == 200, response.content
+
+        assert (
+            mock_rpc_request.call_args.kwargs["table_requests"][0]
+            .columns[1]
+            .aggregation.extrapolation_mode
+            == ExtrapolationMode.EXTRAPOLATION_MODE_SERVER_ONLY
+        )
+
+        # TODO: Ensure server only extrapolation actually gets applied
```

Comment:
Note: we don't store client_sample_rate and
server_sample_rate in EAP correctly when we store_spans in tests. So can't do a proper E2E test yet. Will get it fixed with the EAP team and follow up.